apps/open-ai/src/app/install/service.ts

```diff
@@ -0,0 +1,42 @@
+import { db } from '@/database/client';
+import { Organisation } from '@/database/schema';
+import { inngest } from '@/inngest/client';
+
+type SetupOrganisationParams = {
+  organisationId: string;
+  token: string;
+  sourceOrganizationId: string;
+  region: string;
+};
+
+export const registerOrganisation = async ({
+  organisationId,
+  token,
+  sourceOrganizationId,
+  region,
+}: SetupOrganisationParams) => {
+  const [organisation] = await db
```

Comment:
We need to do a check to see if organizationId & token given by user is valid.
As their is no API endpoints designed for this, we can fetch user here.
```suggestion
  // api call to check that token & sourceOrganizaton are correct
  await getUsers(token, sourceOrganizationId)

  const [organisation] = await db
```

Reply:
Done!